pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/MessageDeduplication.java

```diff
@@ -337,12 +338,23 @@ public MessageDupStatus isDuplicate(PublishContext publishContext, ByteBuf heade
             publishContext.setOriginalHighestSequenceId(highestSequenceId);
             headersAndPayload.readerIndex(readerIndex);
         }
-
+        long chunkID = 0;
+        if (publishContext.isChunked()) {
+            if (md == null) {
+                headersAndPayload.markReaderIndex();
```

Comment:
This code is duplicated with lines 331-339. We could reuse the code.

Reply:
This is for performance reasons. We shouldn’t parse metadata every time.